eth/downloader/downloader.go

```diff
@@ -65,12 +62,11 @@ var (
 	maxHeadersProcess = 2048      // Number of header download results to import at once into the chain
 	maxResultsProcess = 2048      // Number of content download results to import at once into the chain
 
-	fsHeaderCheckFrequency        = 100  // Verification frequency of the downloaded headers during fast sync
-	fsHeaderSafetyNet             = 2048 // Number of headers to discard in case a chain violation is detected
-	fsHeaderForceVerify           = 24   // Number of headers to verify before and after the pivot to accept it
-	fsPivotInterval               = 512  // Number of headers out of which to randomize the pivot point
-	fsMinFullBlocks               = 1024 // Number of blocks to retrieve fully even in fast sync
-	fsCriticalTrials       uint32 = 10   // Number of times to retry in the cricical section before bailing
+	fsHeaderCheckFrequency = 100             // Verification frequency of the downloaded headers during fast sync
+	fsHeaderSafetyNet      = 2048            // Number of headers to discard in case a chain violation is detected
+	fsHeaderForceVerify    = 24              // Number of headers to verify before and after the pivot to accept it
+	fsHeaderContCheck      = 3 * time.Second // Time interval to check for header continuations during state download
+	fsMinFullBlocks        = 64              // Number of blocks to retrieve fully even in fast sync
 )
```

Comment:
Right, so here's where the fun starts. The downloader gets a major refactoring w/r/t (with regard to) `FastSync` procedure. Notice that `fsCriticalTrials` is gone, and that that was one of the most perniciously nondeterministic tests. This won't be the last you'll hear from me about fast sync and the downloader.